arelle/plugin/validate/ESEF/ESEF_Current/ValidateXbrlFinally.py

```diff
@@ -93,11 +93,24 @@ def validateXbrlFinally(val: ValidateXbrl, *args: Any, **kwargs: Any) -> None:
         assert isinstance(modelXbrl.fileSource.fs, zipfile.ZipFile)
 
         maxMB = float(reportPackageMaxMB)
+        _size: int | None = None
+        _sizeExact = True
         if val.authParam["reportPackageMeasurement"] == "unzipped":
             _size = sum(zi.file_size for zi in modelXbrl.fileSource.fs.infolist())
+            _sizeExact = False
         else:
-            _size = sum(zi.compress_size for zi in modelXbrl.fileSource.fs.infolist())
-            # not usable because zip may be posted or streamed: _size = os.path.getsize(modelXbrl.fileSource.basefile)
+            try:
+                _size = modelXbrl.fileSource.getBytesSize()
+            except Exception:
+                pass
+            if _size is None:
+                _size = sum(zi.compress_size for zi in modelXbrl.fileSource.fs.infolist())
+                _sizeExact = False
```

Comment:
This estimation logic could possibly be incorporated into a `getEstimatesBytesSize`.